fedimint-api/src/bitcoin_rpc.rs

```diff
@@ -5,20 +5,59 @@ use url::Url;
 /// Name of the env value used for passing bitcoind rpc url to modules that need it
 pub const FM_BITCOIND_RPC_ENV: &str = "FM_BITCOIND_RPC";
 
+/// Name of the env value used for passing bitcoind rpc url to modules that need it
+pub const FM_ELECTRUM_RPC_ENV: &str = "FM_ELECTRUM_RPC";
+
 /// Default url that will be used if [`FM_BITCOIND_RPC_ENV`] is not set
 pub const FM_BITCOIND_RPC_DEFAULT_FALLBACK: &str = "http://127.0.0.1:8332";
 
+/// Bitcoin
+pub enum BitcoindRpcBackend {
+    /// Bitcoin Core RPC port
+    Bitcoind(Url),
+    /// Electrum RPC port
+    Electrum(Url),
+}
+
 /// Get the value of url the module would use by reading it from process environemnt
 ///
 /// Should be used in test code only to mimick prod code behavior
-pub fn read_bitcoin_rpc_env_from_global_env() -> anyhow::Result<Url> {
-    fm_bitcoind_rpc_env_value_to_url(std::env::var_os(FM_BITCOIND_RPC_ENV).as_deref())
+pub fn read_bitcoin_backend_from_global_env() -> anyhow::Result<BitcoindRpcBackend> {
+    select_bitcoin_backend_from_envs(
+        std::env::var_os(FM_BITCOIND_RPC_ENV).as_deref(),
+        std::env::var_os(FM_ELECTRUM_RPC_ENV).as_deref(),
+    )
+}
```

Comment:
Why make this two different env vars instead of e.g. `electrum://ip:port`, `electrumssl://ip:port` or `bitcoind://ip:port`?

Reply:
@elsirion Not sure. Then we kind of invent new URL schemas which I don't think are official. Or are they?

Reply:
I consider electrum support to be a bit of a hack/wheelchairs, and maybe it's better to have a separate env? If someone meant bitcoind, it shouldn't possibly switch to electrum etc.

I don't know. Mixed feelings. One one hand two env vars are kind of meh, on the other - cramming them together also kind of meh. :D